src/cirrus/github_tools.py

```diff
@@ -41,6 +42,42 @@ def branch_status(branch_name):
     state = resp.json()['state']
     return state
 
+def current_branch_mark_status(repo_dir, state):
+    """
+    _current_branch_mark_status_
+
+    Mark the CI status of the current branch.
+
+    :param repo_dir: directory of git repository
+    :param state: state of the last test run, such as "success" or "failure"
+
+    """
+
+    config = load_configuration()
+    token = get_github_auth()[1]
+    sha = git.Repo(repo_dir).head.commit.hexsha
+    
+    url = "https://api.github.com/repos/{org}/{repo}/commits/{sha}".format(
+        org=config.organisation_name(),
+        repo=config.package_name(),
+        sha=sha
+    )
+
+    headers = {
+        'Authorization': 'token {0}'.format(token),
+        'Content-Type': 'application/json'
+    }
+
+    data = json.dumps(
+        {
+            "state": state,
+            "description": "State after cirrus check.",
+            "context": "cirrus"
```

Comment:
I think that we may need to make the context value configurable. I may be wrong, but I believe that branch protection is generally configured to wait on a status from a specific context string.